packages/did-comm/src/transports/transports.ts

```diff
@@ -100,17 +100,43 @@ export class DIDCommHttpTransport extends AbstractDIDCommTransport {
 
   /** {@inheritdoc AbstractDIDCommTransport.isServiceSupported} */
   isServiceSupported(service: any) {
-    // FIXME: TODO: addtionally handle serviceEndpoint objects in did docs
+    // serviceEndpoint can be a string, a ServiceEndpoint object, or an array of strings or ServiceEndpoint objects
     return (
-      typeof service.serviceEndpoint === 'string' &&
-      (service.serviceEndpoint.startsWith('http://') || service.serviceEndpoint.startsWith('https://'))
+      (typeof service.serviceEndpoint === 'string'  && (
+        service.serviceEndpoint.startsWith('http://') || service.serviceEndpoint.startsWith('https://')
+      ))
+      || 
+      (service.serviceEndpoint.uri &&  typeof service.serviceEndpoint.uri === 'string' && (
+        service.serviceEndpoint.uri.startsWith('http://') || service.serviceEndpoint.uri.startsWith('https://')
+      ))
+      || 
+      (service.serviceEndpoint.length > 0 &&  typeof service.serviceEndpoint[0] === 'string' && (
+        service.serviceEndpoint[0].startsWith('http://') || service.serviceEndpoint[0].startsWith('https://')
+      ))
+      || 
+      (service.serviceEndpoint.length > 0 &&  typeof service.serviceEndpoint[0].uri === 'string' && (
+        service.serviceEndpoint[0].uri.startsWith('http://') || service.serviceEndpoint[0].uri.startsWith('https://')
+      ))
     )
   }
 
   /** {@inheritdoc AbstractDIDCommTransport.send} */
   async send(service: any, message: string): Promise<IDIDCommTransportResult> {
+    let serviceEndpointUrl = ''
+    if (typeof service.serviceEndpoint === 'string') {
+      serviceEndpointUrl = service.serviceEndpoint
+    } else if (service.serviceEndpoint.uri) {
+      serviceEndpointUrl = service.serviceEndpoint.uri
+    } else if (service.serviceEndpoint.length > 0) {
+      if (typeof service.serviceEndpoint[0] === 'string') {
```

Comment:
somehow I'm sure picking just the first element will come back to byte us later, but I'm not sure if this should be treated differently at this point

Reply:
yeah, there's an existing `TODO` about "better strategy for selecting the transport" but it's unclear what that strategy would be

Reply:
Let's leave it as is and solve this when it's an issue for someone and the expectations are more clear